Resources/config/services.xml

```diff
@@ -19,9 +19,14 @@
     </parameters>
 
     <services>
-        <!-- JWT Encoder / Decoder / Default implementation -->
+        <!--  Default JWT Encoder / Decoder -->
+        <service id="lexik_jwt_authentication.encoder.default" class="Lexik\Bundle\JWTAuthenticationBundle\Encoder\DefaultEncoder">
+            <argument type="service" id="lexik_jwt_authentication.jws_provider"/>
+        </service>
+        <!-- JWT Encoder / Decoder / Old implementation (deprecated) -->
         <service id="lexik_jwt_authentication.jwt_encoder" class="%lexik_jwt_authentication.jwt_encoder.class%">
             <argument type="service" id="lexik_jwt_authentication.openssl_key_loader"/>
+            <deprecated>The "%service_id%" is deprecated since version 2.0 and will be removed in 3.0, use "lexik_jwt_authentication.encoder.default" instead.</deprecated>
```

Comment:
Now that I am playing with your updated code, I think I'm going to change my mind :). I think we should remove the 2 deprecated services (`lexik_jwt_authentication.jwt_encoder` & `lexik_jwt_authentication.openssl_key_loader` and their classes. We can simply add this to an `UPGRADE` file. In both cases, the new service works *almost* identically to the old. And unless you're working with these services in some deep way, it will just work.

For me, in the KnpU tutorial, I was able to upgrade the library, change 2 spots to reference `lexik_jwt_authentication.encoder` instead of `lexik_jwt_authentication.jwt_encoder` (which, probably I should have been using anyways), and everything just worked. Even my old JWT's worked out-of-the-box with the new system. Nice job :)